cclib/bridge/cclib2ase.py

```diff
@@ -0,0 +1,243 @@
+        atomnos - one dimensional array-like with atomic numbers.
+        atomcharges - one dimensional array-like with atomic charges.
+        atomspins - one dimensional array-like with atomic spin densities.
+        atommasses - one dimensional array-like with atomic masses.
+    """
+    _check_ase(_found_ase)
+    return Atoms(
+        positions=atomcoords,
+        numbers=atomnos,
+        masses=atommasses,
+        charges=atomcharges,
+        magmoms=atomspins,
+    )
+
+
+def writetraj(filename, ccdata, popname="mulliken", index=None):
+    """Write an ASE Trajectory object from a ccData object.
+
+    We try to write the following properties: atomcoords, atomnos, atomcharges,
+    atomspins, atommasses, scfenergies, grads, moments (dipole only) and
+    freeenergy. No charge or mult is written since ASE calculates them from
+    atomcharges and atomspins.
+
+    When a program do a final single point calculation at the end of an
+    optimization (e.g., ORCA), we'll have unequal number of grads and
+    atomcoords. This means the last geometry in the written .traj will lack
+    forces.
+
+    Some unit conversions are done here, but things are converted back in
+    readtraj/makecclib.
+
+    Inputs:
+        filename - path to traj file to be written.
+        ccdata - an instance of ccData.
+        popname - population analysis to use for atomic partial charges and
+            atomic spin densities. Molecular charge and multiplicity are
+            evaluated from them.
+        index - sequence of integers indicating which atomcoords indices should
+            be exported. By default, all are exported.
+    """
+    _check_ase(_found_ase)
+    traj = Trajectory(filename, "w")
+    for i, atomcoords in enumerate(ccdata.atomcoords):
+        if index is not None and i not in index:
+            continue
+
+        atomspins = None
+        if hasattr(ccdata, "atomspins"):
+            atomspins = ccdata.atomspins[popname]
+        atoms = makease(
+            atomcoords,
+            ccdata.atomnos,
+            ccdata.atomcharges[popname],
+            atomspins,
+            ccdata.atommasses,
+        )
+
+        properties = {}
+        if hasattr(ccdata, "scfenergies"):
+            properties.update({"energy": ccdata.scfenergies[i]})
+        if hasattr(ccdata, "grads"):
+            try:
+                properties.update(
+                    {"forces": -ccdata.grads[i] * units.Hartree / units.Bohr}
+                )
+            except IndexError:
+                pass
+
+        if i == len(ccdata.atomcoords) - 1:  # last geometry
+            if hasattr(ccdata, "moments"):
+                properties.update({"dipole": ccdata.moments[1] * units.Bohr})
+            if hasattr(ccdata, "free_energy"):
+                properties.update(
+                    {"free_energy": ccdata.freeenergy * units.Hartree}
+                )
+
+        traj.write(atoms, **properties)
+
+
+def readtraj(filename):

```

Comment:
Same for this and other methods.

Reply:
I changed both to `write_trajectory` and `read_trajectory`. The other three functions (`_check_ase`, `makease` and `makecclib`) are in line with how they are called in other bridges (for instance, [`cclib2openbabel._check_openbabel`](https://github.com/cclib/cclib/blob/master/cclib/bridge/cclib2openbabel.py#L23), [`cclib2openbabel.makeopenbabel`](https://github.com/cclib/cclib/blob/master/cclib/bridge/cclib2openbabel.py#L50), etc.). Please let me know if this needs to change as well.